src/main.rs

```diff
@@ -432,6 +459,69 @@ fn run_cli(cli: Cli) -> Result<()> {
                 }
             }
         },
+        Command::Attach {
+            citation_key,
+            file,
+            rename,
+            force,
+        } => {
+            // Initialize the file directory path
+            let file_dir = if let Some(file_dir) = cli.file_dir {
+                // at a user-provided path
+                info!(
+                    "Using user-provided file directory '{}'",
+                    file_dir.display()
+                );
+                file_dir
+            } else {
+                // at the default path
+                let default_db_path = data_dir.join("attachments");
+                info!(
+                    "Using default file directory '{}'",
+                    default_db_path.display()
+                );
+
+                default_db_path
+            };
+
+            // Extend with the filename.
+            let (record, row) =
+                get_record_row(&mut record_db, citation_key, &client, &config.on_insert)?
+                    .exists_or_commit_null("Cannot show directory for")?;
+            let Record { canonical, .. } = record;
+            let mut target = file_dir;
+            canonical.path_hash(&mut target);
+            row.commit()?;
+
+            // Try to open the source file first, since this will reduce the number of redundant
+            // errors.
+            let mut source_file = File::open(&file)?;
+
+            if let Some(ext) = rename {
+                if !ext.is_relative() {
+                    bail!("New path must relative");
+                }
+                target.push(ext);
+            } else if let Some(ext) = file.file_name() {
+                target.push(ext);
+            } else {
+                bail!("Source file must not be a directory");
+            }
+
```

Comment:
A more robust solution is to check whether the target directory is a prefix of the final target path.